src/join.jl

```diff
@@ -187,8 +185,9 @@ function naturaljoin(op, left::DNDSparse{I1,D1},
 
         append!(out_domains, overlapping_domains)
     end
-
-    return cache_thunks(DNDSparse{I, D}(out_domains, out_chunks))
+    computed_chunks = map(t -> compute(get_context(), t), out_chunks)
```

Comment:
Not sure this is the correct way to do things, but somehow I think we need to compute things to get the output type without inference.

Reply:
This type of compute will be serial... It will wait for every chunk to be computed.

Just do `fromchunks(out_chunks)` -- that will trigger compute and figure out the output type.

Reply:
ah, I see... Should I do `fromchunks(out_chunks, domains = out_domains)` to keep that information?